### ISSUE TITLE
Support for multiboot protocol among firmwares

### ISSUE THREAD

**Issue by proger** [mist64/xhyve#54]
[Multiboot](http://www.gnu.org/software/grub/manual/multiboot/multiboot.html) is used to boot [Xen](http://wiki.xenproject.org/wiki/Xen_ARM_with_Virtualization_Extensions/Multiboot) and other GNU stuff. [FreeBSD and NetBSD](http://www.syslinux.org/wiki/index.php/Mboot.c32) kernels seem to be multiboot-compliant as well.

Do you have any plans to add multiboot support to xhyve or is there any other simple(r) hack that can be used to boot Xen? Looks like there is [EFI support](http://wiki.xenproject.org/wiki/Xen_EFI) but I'm not sure what your plans towards EFI are.

**Comment by xez** [comment]
Multiboot should be pretty straightforward to implement, just a matter of priorities and spare time.

Do you have any particular use cases for xen in xhyve in mind? The only thing that would work is full PV, not HVM (since bhyve does not yet support nested hardware virtualization).

A more general solution would be EFI in the guest to boot arbitrary operating systems. There is some work in the bhyve community to make that happen. Once this is done this will likely *just work (tm)* in xhyve.

**Comment by proger** [comment]
Yep, I'd like to use PV as it's necessary for [Erlang on Xen](http://erlangonxen.org/).

**Comment by maxogden** [comment]
+1, found this issue trying to boot http://runtimejs.org/ on xhyve. currently doesn't work due to lack of multiboot

**Comment by kfix** [comment]
Doesn't http://github.com/grehan-freebsd/grub2-bhyve should handle Multiboot images?

**Comment by stv0g** [comment]
I started implementing the Multiboot specification v1 here: https://github.com/stv0g/xhyve/tree/multiboot

I would like to use it, to boot [eduOS](https://github.com/RWTH-OS/eduOS).

Grub2 itself is also a valid Multiboot image. So we can chainload Grub2 with it.

**Comment by rickpayne** [comment]
Chalk me up as another one interested in Xen, again for Erlang on Xen...

**Comment by stv0g** [comment]
Multiboot support is almost ready :-) Expect it in a week or two.
I will notify you. Testers are welcome.

Am 2. Mai 2016 15:20:55 MESZ, schrieb Rick Payne <notifications@github.com>:
>Chalk me up as another one interested in Xen, again for Erlang on
>Xen...
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/mist64/xhyve/issues/54#issuecomment-216232469

-- 
Steffen Vogel
Robensstraße 69
52070 Aachen

Mail: post@steffenvogel.de
Mobil: +49 1575 7180927
Web: http://www.steffenvogel.de
Jabber: steffen.vogel@jabber.rwth-aachen.de

**Comment by rickpayne** [comment]
Any progress on this stv0g?
